keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -284,6 +287,48 @@ class SqlVerificationRepository(
     }
   }
 
+  override fun pendingInEnvironment(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String
+  ): Collection<PendingVerification> {
+    return jooq
+      .select(
+        DELIVERY_ARTIFACT.REFERENCE,
+        VERIFICATION_STATE.ARTIFACT_VERSION,
+        VERIFICATION_STATE.VERIFICATION_ID,
+        VERIFICATION_STATE.STATUS,
+        VERIFICATION_STATE.STARTED_AT,
+        VERIFICATION_STATE.ENDED_AT,
+        VERIFICATION_STATE.METADATA,
+      )
+      .from(VERIFICATION_STATE)
+      .join(ENVIRONMENT)
+      .on(ENVIRONMENT.UID.eq(VERIFICATION_STATE.ENVIRONMENT_UID))
+      .and(ENVIRONMENT.NAME.eq(environmentName))
+      .join(DELIVERY_CONFIG)
+      .on(DELIVERY_CONFIG.UID.eq(ENVIRONMENT.DELIVERY_CONFIG_UID))
+      .and(DELIVERY_CONFIG.NAME.eq(deliveryConfig.name))
+      .join(DELIVERY_ARTIFACT)
+      .on(DELIVERY_ARTIFACT.UID.eq(VERIFICATION_STATE.ARTIFACT_UID))
+      .where(VERIFICATION_STATE.STATUS.eq(PENDING))
```

Comment:
I changed the query in this repository method to only return `PENDING` verifications. We may have old ones that are `NOT_EVALUATED` and _never_ launched, but they should not block new verifications from happening, or prevent promotion of a new artifact version.

Reply:
We need to remember that `PENDING` now has explicit semantics in keel, it means "actively running". If we ever add an additional state that is also true when actively running, it could bite us if we don't remember to update this query.

Might be a good idea to abstract out the where clause into a function to make this more explicit, e.g.:

```kotlin
fun Select.whereVerificationIsRunning() ...
```